python/pyspark/serializers.py

```diff
@@ -259,11 +261,13 @@ def load_stream(self, stream):
         """
         Deserialize ArrowRecordBatches to an Arrow table and return as a list of pandas.Series.
         """
+        from pyspark.sql.types import _check_dataframe_localize_timestamps
         import pyarrow as pa
         reader = pa.open_stream(stream)
         for batch in reader:
-            table = pa.Table.from_batches([batch])
-            yield [c.to_pandas() for c in table.itercolumns()]
+            # NOTE: changed from pa.Columns.to_pandas, timezone issue in conversion fixed in 0.7.1
+            pdf = _check_dataframe_localize_timestamps(batch.to_pandas())
+            yield [c for _, c in pdf.iteritems()]
```

Comment:
After running some tests, this change does not significantly degrade performance, but there seems to be a small difference.  cc @ueshin 

I ran various columns of random data through a `pandas_udf` repeatedly with and without this change.  Test was in local mode with default Spark conf, looking at min wall clock time of 10 loops

before change: 2.595558
after change: 2.681813

Do you think the difference here is acceptable for now until arrow is upgraded and we can look into again?

Reply:
I ran your script in my local, too.

- before change: 
  - mean: 2.605722
  - min: 2.502404
  - max: 3.045294
- after change:
  - mean: 2.626306
  - min: 2.341781
  - max: 2.742432

I think it's okay to use this workaround.